abi/valuesSingle.go

```diff
@@ -0,0 +1,106 @@
+package abi
+
+import "math/big"
+
+// U8Value is a wrapper for uint8
+type U8Value struct {
+	Value uint8
+}
+
+// U16Value is a wrapper for uint16
+type U16Value struct {
+	Value uint16
+}
+
+// U32Value is a wrapper for uint32
+type U32Value struct {
+	Value uint32
+}
+
+// U64Value is a wrapper for uint64
+type U64Value struct {
+	Value uint64
+}
+
+// I8Value is a wrapper for int8
+type I8Value struct {
+	Value int8
+}
+
+// I16Value is a wrapper for int16
+type I16Value struct {
+	Value int16
+}
+
+// I32Value is a wrapper for int32
+type I32Value struct {
+	Value int32
+}
+
+// I64Value is a wrapper for int64
+type I64Value struct {
+	Value int64
+}
+
+// BigIntValue is a wrapper for a big integer (unsigned)
+type BigUIntValue struct {
+	Value *big.Int
+}
+
+// BigIntValue is a wrapper for a big integer (signed)
+type BigIntValue struct {
+	Value *big.Int
+}
+
+// AddressValue is a wrapper for an address
+type AddressValue struct {
+	Value []byte
+}
+
+// BytesValue is a wrapper for a byte slice
+type BytesValue struct {
+	Value []byte
+}
+
+// StringValue is a wrapper for a string
+type StringValue struct {
+	Value string
+}
+
+// BoolValue is a wrapper for a boolean
+type BoolValue struct {
+	Value bool
+}
+
+// OptionValue is a wrapper for an optional value
```

Comment:
...for an `option` value

`optional` is different thing :D

Reply:
Indeed, fixed.